### PR TITLE
Fixes #6386, Use copies instead of symbolic files

### LINKED ISSUE #6386: Make Windows build stop clobbering tools/shared/cwrap_common.py
Because this file is implemented as a symlink, in Windows we copy the source over it as part of the build process. This dirties the working copy, and is generally annoying. It would be good to work out a way to make this not happen.

CC @peterjc123

### PR DESCRIPTION
**Pull request by peterjc123**

### CODE REVIEW

**Review by ezyang — DISMISSED** [review]
Won't your working copy still be clobbered after running setup.py?

**Comment by peterjc123** [comment]
@ezyang No, it is listed in `.gitignore`. So the working copy won't change.

**Comment by ezyang** [comment]
That's not how gitignore works. If a file is already committed in the repo, gitignore has no affect on whether or not updates show up or not.

**Comment by peterjc123** [comment]
@ezyang So would you please tell me what should I do?

File: setup.py [hunk 1]
```diff
@@ -105,6 +105,17 @@
...
 
+
+# Use copies instead of symbolic files.
+# Windows has very poor support for them.
+sym_files = ['tools/shared/cwrap_common.py']
+orig_files = ['aten/src/ATen/common_with_cwrap.py']
+for sym_file, orig_file in zip(sym_files, orig_files):
+    if os.path.exists(sym_file):
+        os.remove(sym_file)
+    shutil.copyfile(orig_file, sym_file)
```
Comment (apaszke) [inline setup.py]: We probably don't want to do this every time `setup.py` is ran. It should be part of the command that needs it (`build_deps`?)

**Review by apaszke — CHANGES_REQUESTED** [review]

**Comment by ezyang** [comment]
I think one concrete course of action that will probably work is to delete the symlink `tools/shared/cwrap_common.py`, and then do some sort of copy to fill it in (as I mentioned in #6386). The primary downside is that timestamp on the file would keep changing, but given that it's Python I don't think it should matter that much; at least, the Windows code suggests as much. @apaszke I don't know what you think about that, but I don't think it should cause too much trouble.

**Comment by apaszke** [comment]
@ezyang if you put things in `setup.py` then they will run no matter what command you execute. This include things like `egginfo`, which should only print some information, and we should avoid side effects in this case.

**Review by apaszke — APPROVED** [review]

**Pull request merged**